File: Ruby/day1_puzzle1.rb
list1, list2 = File
  .read("../input/day1.txt")
  .split("\n")
  .map { |l| l.split("   ") }
  .transpose

list1.map!(&:to_i).sort!
list2.map!(&:to_i).sort!

result = list1.zip(list2).map { |x1, x2| (x1.to_i - x2.to_i).abs }.sum

puts result

File: python/day1_puzzle1.py
with open("../_input/day1.txt", "r") as f:
    list1, list2 = zip(*[line.strip().split("   ") for line in f.readlines()])

list1 = sorted(map(int, list1))
list2 = sorted(map(int, list2))

result = sum(abs(x1 - x2) for x1, x2 in zip(list1, list2))

print(result)
